pkg/install/stack/kubermatic-master/stack.go

```diff
@@ -368,164 +367,35 @@ func showDNSSettings(ctx context.Context, logger *logrus.Entry, kubeClient ctrlr
-		logger.Warn("Please check the DaemonSet and, if necessary, reconfigure the")
-		logger.Warn("nginx-ingress-controller Helm chart. Re-run the installer to apply")
-		logger.Warn("updated configuration afterwards.")
-		return
-	}
-
-	logger.Info("The nginx-ingress-controller pods have been rolled out in your cluster.")
-	logger.Info("")
-
-	logger.Infof("  %-50s    IP / Hostname", "Pod")
-	for _, target := range targets {
-		logger.Infof("  %-50s    %s", target.pod, target.prefererdTarget())
-	}
-
-	domain := opt.KubermaticConfiguration.Spec.Ingress.Domain
-	rand := targets[rand.Intn(len(targets))]
-
-	logger.Info("")
-	logger.Info("Choose a single target for your DNS or configure an external LoadBalancer")
-	logger.Info("to balance between all targets listed above. For a basic setup, choose a")
-	logger.Infof("random target from above, for example %s.", rand.prefererdTarget())
-	logger.Infof("Then ensure your DNS settings for %q include the following records:", domain)
-	logger.Info("")
-
-	if rand.dns != "" {
-		logger.Infof("   %s.    IN  CNAME  %s.", domain, rand.dns)
-		logger.Infof("   *.%s.  IN  CNAME  %s.", domain, rand.dns)
-	} else {
-		logger.Infof("   %s.    IN  A  %s", domain, rand.ip)
-		logger.Infof("   *.%s.  IN  A  %s", domain, rand.ip)
-	}
-
-	logger.Info("")
-}
-
-func showLoadBalancerDNSSettings(ctx context.Context, logger *logrus.Entry, kubeClient ctrlruntimeclient.Client, opt stack.DeployOptions) {
-	svcName := types.NamespacedName{
-		Namespace: NginxIngressControllerNamespace,
-		Name:      "nginx-ingress-controller",
+	ingressName := types.NamespacedName{
+		Namespace: opt.KubermaticConfiguration.Namespace,
+		Name:      operatorcommon.IngressName,
 	}
 
-	logger.Debugf("Waiting for %q to be ready…", svcName)
+	logger.WithField("ingress", ingressName).Debug("Waiting for Ingress to be ready…")
 
-	var ingresses []corev1.LoadBalancerIngress
-	err := wait.PollUntilContextTimeout(ctx, 5*time.Second, 3*time.Minute, true, func(ctx context.Context) (bool, error) {
-		svc := corev1.Service{}
-		if err := kubeClient.Get(ctx, svcName, &svc); err != nil {
+	var ingresses []networkingv1.IngressLoadBalancerIngress
+	err := wait.PollUntilContextTimeout(ctx, 3*time.Second, 3*time.Minute, true, func(ctx context.Context) (bool, error) {
+		ingress := networkingv1.Ingress{}
+		if err := kubeClient.Get(ctx, ingressName, &ingress); err != nil {
 			return false, err
 		}
 
-		ingresses = svc.Status.LoadBalancer.Ingress
+		ingresses = ingress.Status.LoadBalancer.Ingress
 
 		return len(ingresses) > 0, nil
 	})
 	if err != nil {
-		logger.Warnf("Timed out waiting for the LoadBalancer service %q to become ready.", svcName)
+		logger.Warn("Timed out waiting for the Ingress to become ready.")
 		logger.Warn("Please check the Service and, if necessary, reconfigure the")
 		logger.Warn("nginx-ingress-controller Helm chart. Re-run the installer to apply")
 		logger.Warn("updated configuration afterwards.")
```

Comment:
I think it would be neat to give the `Ingress` resource name here, so users can check out the resource for any problems easily.

In addition, the log warnings below don't seem to be correct anymore:

> Please check the Service

We no longer care about the service, should IMHO also be `Ingress`.

Reply:
I know what you mean and I saw that when I made the PR. I decided to not change the wording because a) I'm lazy and b) the code above is still used when using our nginx. And if something is not working, I wanted to direct the user to check the "ingress tool you're using" (the "Service") and not look at the Ingress. I assumed the Ingress will not show any issues, will it?

But I have no super strong opinions either way.

Reply:
But when nginx is broken, it will not be the Ingress that is malfunctioning, but the LB. So I do not expect any "Failed to create loadBalancer: ARN bla bla SKU AWS bla bla" error message from the aws CCM on the Ingress. Why would the CCM "broadcast" this to all Ingresses in a cluster?

Reply:
Fair enough. Let me not block too much on this, although I think the language should be improved at some point.